src/System.Private.CoreLib/shared/System/ComponentModel/DefaultValueAttribute.cs

```diff
@@ -44,14 +56,32 @@ public DefaultValueAttribute(Type type, string value)
                 {
                     _value = TimeSpan.Parse(value);
                 }
-                else
+                else if (type.Module == typeof(string).Module)
```

Comment:
This condition looks suspect. There are intrinsic type converters for many random types in CoreLib: https://github.com/dotnet/corefx/blob/e0ba7aa8026280ee3571179cc06431baf1dfaaac/src/System.ComponentModel.TypeConverter/src/System/ComponentModel/ReflectTypeDescriptionProvider.cs#L103

Reply:
Should we always use the TypeConverters when they are available?

Reply:
i think that the idea comes from https://referencesource.microsoft.com/#System/compmod/system/componentmodel/DefaultValueAttribute.cs,b956ede3a6b29f1f,references same "intrinsics" are present also on .net full?

Reply:
> Should we just always use the TypeConverters for everything when they are available?

If custom converter is not defined GetConverter return a "default type converter". Do you mean check if return something different than "default type converter" and in that case fallback to `_value = Convert.ChangeType(value, type, CultureInfo.InvariantCulture);` ?

Reply:
I think the upstack type converters should take care of all conversions. I do not think we need any fallbacks once the upstack type converter is called.

Reply:
Yeah, that's what desktop does.  My gist was merely trying to preserve the existing "fast path" but I can see how it would be broken if something in CoreLib had an intrinsic converter in TypeConverter and was missing from Convert.ChangeType.

@jkotas do you have any concerns around perf of the TypeConverter that might warrant a fast path?

Reply:
No concerns

Reply:
Do i remove all fast path and use only TypeConverter?

Reply:
> If type converter is present, use it

As i said above if converter is not present we get default type converter(not null). Do you mean che if GetConverter return something different than default right?

Reply:
Ah ok trimmed assemblies understood thank's!

Reply:
done